inst/pages/98_exercises.qmd

```diff
@@ -568,42 +568,183 @@ colData(tse)$Barcode_full_length
 
 ### Subsetting
 
+In this exercise, we'll go over the basics of subsetting a TreeSE object.
+Keep in mind that it's good practice to always keep the original data intact.
+Therefore, we suggest creating a new TreeSE object whenever you subset.
+
+Please have a try at the following exercises. 
+
 1. Subset the TreeSE object to specific samples.  
+
+```{r}
+#| label: subsetting-ex1
+#| eval: FALSE
+#| code-fold: true
+#| code-summary: "Show solution"
+
+tse_subset_by_sample <- tse[ , tse$SampleType %in% 
+                               c("Feces", "Skin", "Tongue")]
+
+unique(tse_subset_by_sample$SampleType)
+```   
+
 2. Subset the TreeSE object to specific features.  
+
+```{r}
+#| label: subsetting-ex2
+#| eval: FALSE
+#| code-fold: true
+#| code-summary: "Show solution"
+
+tse_subset_by_sample_and_feature <- tse[rowData(tse)$Phylum %in% c("Crenarchaeota", 
+                                                       "Euryarchaeota",
+                                                       "Actinobacteria")
+                             ,]
+
+dim(tse_subset_by_sample_and_feature)
+```   
+
 3. Subset the TreeSE object to specific samples and features.  
 
+3.1. Find some samples and features to subset with.
+
+```{r}
+#| label: subsetting-ex3.1
+#| eval: FALSE
+#| code-fold: true
+#| code-summary: "Show solution"
+
+head(unique(rowData(tse)$Phylum)) # Features
+
+unique(tse$SampleType) # Samples
+```
+
+3.2. Subset the TreeSE with your found samples and features.
+
+```{r}
+#| label: subsetting-ex3.2
+#| eval: FALSE
+#| code-fold: true
+#| code-summary: "Show solution"
+
+features <- c("Crenarchaeota","Euryarchaeota","Actinobacteria")
+samples <- c("Feces", "Skin", "Tongue")
+
+tse_subset_by_feature <- tse[rowData(tse)$Phylum %in% features
+                             ,
+                             tse$SampleType %in% samples]
+
+unique(rowData(tse_subset_by_feature)$Phylum)
```

Comment:
this seems replicate what is on earlier rows. Can we just keep this example (and remove the other that are almost duplicates)?

Reply:
This example was originally meant to make the user familiar with sample and row subsetting one at a time, and then proceeding to combine the two together. I thought it would be easier to digest one step at a time.

Reply:
I think keeping all three of the subsets can be good to show different ways to subset. If someone's not too familiar with R they might not be aware of these simple subsetting methods. It's true however that if people have used R before, they will likely skip the first two questiond.

Reply:
Ok just see what works best. We should assume some basic R skills from our users, hence it is good to consider the balance.